incubator/zalenium/templates/deployment.yaml

```diff
@@ -0,0 +1,139 @@
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.hub.replicaCount }}
+  strategy:
+    type: {{ default "RollingUpdate" .Values.hub.strategyType }}
+  selector:
+    matchLabels:
+      app: {{ $fullName }}
+      zalenium: {{ .Values.hub.name }}
+  template:
+    metadata:
+      labels:
+        app: {{ $fullName }}
+        zalenium: {{ .Values.hub.name }}
+      {{- with .Values.hub.annotations }}
+        annotations:
+{{ toYaml .Values.hub.annotations | indent 8 }}
+      {{- end }}
+    spec:
+      initContainers:
+        - name: zalenium-prepare
+          image: dosel/zalenium:latest
+          command: ['sh', '-c', 'sudo chown -R seluser:seluser /home/seluser/videos']
+          volumeMounts:
+            - name: {{ $fullName }}-videos
+              mountPath: /home/seluser/videos
+      serviceAccountName: {{ template "serviceAccountName" . }}
+      containers:
+        - name: {{ template "name" . }}-{{ .Values.hub.name }}
+          image: "{{ .Values.hub.image.repository }}:{{ .Values.hub.image.tag }}"
+          imagePullPolicy: {{ .Values.hub.image.pullPolicy }}
+          ports:
+            - containerPort: {{ .Values.hub.port }}
+              protocol: TCP
+          {{- if .Values.hub.livenessProbe.enabled }}
+          livenessProbe:
+            httpGet:
+              path: /grid/console
+              port: {{ .Values.hub.port }}
+            initialDelaySeconds: 90
+            periodSeconds: 5
+          {{- end }}
+          {{- if .Values.hub.readinessProbe.enabled }}
+          readinessProbe:
+            httpGet:
+              path: /grid/console
+              port: {{ .Values.hub.port }}
+            timeoutSeconds: 1
+          {{- end }}
+          resources:
+{{ toYaml .Values.hub.resources | indent 12 }}
+          env:
+            - name: ZALENIUM_KUBERNETES_CPU_REQUEST
+              value: {{ .Values.hub.cpuRequest }}
+            - name: ZALENIUM_KUBERNETES_CPU_LIMIT
+              value: {{ .Values.hub.cpuLimit }}
+            - name: ZALENIUM_KUBERNETES_MEMORY_REQUEST
+              value: {{ .Values.hub.memRequest }}
+            {{- if eq true .Values.hub.sauceLabsEnabled }}
+            - name: SAUCE_USERNAME
+              value: {{ .Values.hub.sauceUserName }}
+            - name: SAUCE_ACCESS_KEY
+              value: {{ .Values.hub.sauceAccessKey }}
+            {{- end }}
+            {{- if eq true .Values.hub.browserStackEnabled }}
+            - name: BROWSER_STACK_USER
+              value: {{ .Values.hub.browserStackUser }}
+            - name: BROWSER_STACK_KEY
+              value: {{ .Values.hub.browserStackKey }}
+            {{- end }}
+            {{- if eq true .Values.hub.testingBotEnabled }}
+            - name: TESTINGBOT_KEY
+              value: {{ .Values.hub.testingBotKey }}
+            - name: TESTINGBOT_SECRET
+              value: {{ .Values.hub.testingBotSecret }}
+            {{- end }}
+          args:
```

Comment:
args can also map from`values.yaml`

```
args:
{{ toYaml .Values.args | indent 4 }}
```

https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#use-environment-variables-to-define-arguments

Reply:
I used to like that, but it will not run properly that it should run. To do this will help you to set each value as needed

Reply:
use `--set key=value,key..n=value..n` to set each args! If you use value.yaml, it will be hard-coded for each value and can not be set for each value by the other.